### PR TITLE
Implemented auto leave threshold features

### LINKED ISSUE #42: leave if participants in a meeting are lower than a threshold
sometimes in a meeting there are kids afk and stay in the meeting for a long time after its over , so could you please add an option where it will auto leave if there are less than 'x'  participants( or 'z' percent of total participants )remaining  after ' y' time has passed in a meeting

### PR DESCRIPTION
**Pull request by jevonmao**
Added values in config.json for user to use a number of people threshold or percentage of people threshold for the auto leave feature. Thanks @TobiasPankner. Close #42

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
 Always joins the newest meeting and leaves either after a specified time, if you are the last person in the meeting or only if a new one is available (see [Configuration options](#configuration-options) for more information).
 I also made a short tutorial video on how to setup the bot: https://youtu.be/YgkSOqfIjf4
 
+## Prerequisites
+
+- Python3 ([Download](https://www.python.org/downloads/))
+
+## Configuration options
```
Comment (TobiasPankner) [inline README.md:20]: unecessary change

File: README.md [hunk 2]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
-## Prerequisites  
-  
- - Python3 ([Download](https://www.python.org/downloads/))  
-   
-## Configuration options  
-  
 - **email/password:**  
-The email/password of your Microsoft account (can be left empty if you don't want to automatically login)  
+  The email/password of your Microsoft account (can be left empty if you don't want to automatically login)
```
Comment (TobiasPankner) [inline README.md:23]: unecessary change

File: README.md [hunk 3]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **run_at_time:**  
-Time to start the script at. Input is a string of the hour and minute in 24h format, if you want it to start immediately leave this empty. 
-If a time before the current time is given, the next day is used. Also make sure that you entered your email and password.
-For example, if you want the script to start searching meetings at 6 in the morning on the next day, you would input `06:00` in the config.
+  Time to start the script at. Input is a string of the hour and minute in 24h format, if you want it to start immediately leave this empty.
+  If a time before the current time is given, the next day is used. Also make sure that you entered your email and password.
+  For example, if you want the script to start searching meetings at 6 in the morning on the next day, you would input `06:00` in the config.
```
Comment (TobiasPankner) [inline README.md:28]: unecessary change

File: README.md [hunk 4]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **meeting_mode:**
-Change which meetings should be joined. Modes 1, 2 and 3 are available.  
-`1` Both channel and calendar meetings  
-`2` Only channel meetings  
-`3` Only calendar meetings  
+  Change which meetings should be joined. Modes 1, 2 and 3 are available.  
+  `1` Both channel and calendar meetings  
+  `2` Only channel meetings  
+  `3` Only calendar meetings
```
Comment (TobiasPankner) [inline README.md:34]: unecessary change

File: README.md [hunk 5]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **organisation_num:**
-If your Teams account is in multiple organisations, as seen in the example below, change the organisation_num to the number of the list item (counting starts from 1)
-<img width="30%" src="https://imgur.com/CWpK4wk.png">
+  If your Teams account is in multiple organisations, as seen in the example below, change the organisation_num to the number of the list item (counting starts from 1)
+  <img width="30%" src="https://imgur.com/CWpK4wk.png">
```
Comment (TobiasPankner) [inline README.md:38]: unecessary change

File: README.md [hunk 6]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **random_delay:**
-If true, adds a random delay (10s-30s) before joining a meeting. Can be useful so the bot seems more "human like".
+  If true, adds a random delay (10s-30s) before joining a meeting. Can be useful so the bot seems more "human like".
```
Comment (TobiasPankner) [inline README.md:41]: unecessary change

File: README.md [hunk 7]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **check_interval:**
-The amount of seconds to wait before checking for meetings again. Only integer numbers greater than 1 are allowed.
+  The amount of seconds to wait before checking for meetings again. Only integer numbers greater than 1 are allowed.
```
Comment (TobiasPankner) [inline README.md:44]: unecessary change

File: README.md [hunk 8]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
+
+- **leave_threshold_number: (Must enable leave_if_last for this to work)**
+  Sets the threshold number of people in the meeting before auto leaving. If number of people in meeting is less than this threshold, program will leave.
+
+- **leave_threshold_percentage: (Must enable leave_if_last for this to work)**
+  Sets the threshold percentage of people still in the meeting before auto leaving.
 
 - **leave_if_last:**
-If true, leaves the meeting if you are the last person in it.
+  If true, leaves the meeting if you are the last person in it.
```
Comment (TobiasPankner) [inline README.md:56]: unecessary change

File: README.md [hunk 9]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **pause_search:**
-If true, doesn't search for new meetings while there is one active. Keep in mind to set auto_leave_after_min or leave_if_last,
-otherwise the bot will not search for meetings again.
+  If true, doesn't search for new meetings while there is one active. Keep in mind to set auto_leave_after_min or leave_if_last,
+  otherwise the bot will not search for meetings again.
```
Comment (TobiasPankner) [inline README.md:60]: unecessary change

File: README.md [hunk 10]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **headless:**
-If true, runs Chrome in headless mode (does not open GUI window and runs in background).
+  If true, runs Chrome in headless mode (does not open GUI window and runs in background).
```
Comment (TobiasPankner) [inline README.md:63]: unecessary change

File: README.md [hunk 11]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **mute_audio:**
-If true, mutes all the sounds.
+  If true, mutes all the sounds.
```
Comment (TobiasPankner) [inline README.md:66]: unecessary change

File: README.md [hunk 12]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **chrome_type:**
-Valid options: `google-chrome`, `chromium`, `msedge`. By default, google chrome is used, but the script can also be used with Chromium or Microsoft Edge.
+  Valid options: `google-chrome`, `chromium`, `msedge`. By default, google chrome is used, but the script can also be used with Chromium or Microsoft Edge.
```
Comment (TobiasPankner) [inline README.md:69]: unecessary change

File: README.md [hunk 13]
```diff
@@ -12,59 +13,65 @@ Automatically turns off your microphone and camera before joining. Automatic log
...
 
 - **blacklist:**
-A list of Teams and their channels to ignore. Meetings ocurring in these channels w**ill not be joined.
-If you have a Team called "Test1" and, within that, two channels called "General" and "Channel1" and you don't want to join meetings in the "General" Channel:
+  A list of Teams and their channels to ignore. Meetings ocurring in these channels w\*\*ill not be joined.
+  If you have a Team called "Test1" and, within that, two channels called "General" and "Channel1" and you don't want to join meetings in the "General" Channel:
+
```
Comment (TobiasPankner) [inline README.md:74]: unecessary change

File: README.md [hunk 14]
```diff
@@ -77,16 +84,16 @@ If you have a Team called "Test1" and, within that, two channels called "General
 ```
 
 - **blacklist_meeting_re:**
-If calendar meeting title matches a regular expression, it goes to blacklist.
-Leave empty to attend to all meetings.
+  If calendar meeting title matches a regular expression, it goes to blacklist.
+  Leave empty to attend to all meetings.
```
Comment (TobiasPankner) [inline README.md:88]: unecessary change

File: README.md [hunk 15]
```diff
@@ -77,16 +84,16 @@ If you have a Team called "Test1" and, within that, two channels called "General
...
- 3. Install dependencies:   ```pip install -r requirements.txt```
- 4. Run [auto_joiner.py](auto_joiner.py): `python auto_joiner.py`
- 5. After starting, teams might be in Grid view, if this is the case change the view to list [(How to do)](https://support.microsoft.com/en-us/office/view-and-organize-your-teams-b9dd0d8c-243a-43a4-9501-ec8017fec32e)
-<img src="https://i.imgur.com/GODoJYf.png?2" width="300" height="245" />
+1.  Rename the [config.json.example](config.json.example) file to "config.json"
+2.  Edit the "config.json" file to fit your preferences (optional)
+3.  Install dependencies: `pip install -r requirements.txt`
+4.  Run [auto_joiner.py](auto_joiner.py): `python auto_joiner.py`
+5.  After starting, teams might be in Grid view, if this is the case change the view to list [(How to do)](https://support.microsoft.com/en-us/office/view-and-organize-your-teams-b9dd0d8c-243a-43a4-9501-ec8017fec32e)
+    <img src="https://i.imgur.com/GODoJYf.png?2" width="300" height="245" />
```
Comment (TobiasPankner) [inline README.md:99]: unecessary change

File: config.json.example [hunk 16]
```diff
@@ -5,15 +5,19 @@
   "run_at_time": "",
 
   "meeting_mode": 1,
-  "organisation_num" : 1,
+  "organisation_num": 1,
```
Comment (TobiasPankner) [inline config.json.example:8]: unecessary change

File: config.json.example [hunk 17]
```diff
@@ -5,15 +5,19 @@
...
   "auto_leave_after_min": -1,
+
   "leave_if_last": false,
+  "leave_threshold_number": "",
+  "leave_threshold_percentage": "",
+
   "pause_search": false,
 
-  "headless" : false,
+  "headless": false,
```
Comment (TobiasPankner) [inline config.json.example:20]: unecessary change

File: config.json.example [hunk 18]
```diff
@@ -25,10 +29,9 @@
         "Second Channel to blacklist"
       ]
     },
-     {
+    {
       "team_name": "Blacklist whole team",
-      "channel_names": [
-      ]
+      "channel_names": []
```
Comment (TobiasPankner) [inline config.json.example:34]: unecessary change

**Review by TobiasPankner — CHANGES_REQUESTED** [review]
Please remove the unecessary format changes. Otherwise I really like this change and will merge it. 
Thank you.

**Comment by TobiasPankner** [comment]
Please also fix the formatting issues for the Readme file.

**Comment by jevonmao** [comment]
I tried to fix the formatting issues. Not sure if it worked, please let me know if further revision is required.

**Comment by TobiasPankner** [comment]
Thanks for contributing!

**Pull request merged**